### DISCUSSION TITLE
S-parameters from AC simulation

### DISCUSSION THREAD

**Discussion by tomhajjar** [ra3xdh/qucs_s#569] [Q&A]
I ran into an issue converting a Qucs project to Qucs-S/ngspice. Although irrelevant now that Qucs-S does s-parameters, it could make a good Template for learning.

1. Unless I added "Z0=50" into the .PARAM statement AND the Nutmeg equation block the simulation wouldn't run.
2. Other than adding I1/I2, is there a simpler way to get V1/V2 currents into the equations?

I probably could have used the old S-parameter probes from years ago but didn't... 

![2024-02-15_164545](https://github.com/ra3xdh/qucs_s/assets/102838604/fb996ae2-8d11-4d7f-8768-ad455bffe7e7)

**Comment by ra3xdh** [comment]
> Unless I added "Z0=50" into the .PARAM statement AND the Nutmeg equation block the simulation wouldn't run.

It is not a bug. Nutmeg postprocessor is not aware about .PARAM parameters. You have to duplicate the parameters. 

> Other than adding I1/I2, is there a simpler way to get V1/V2 currents into the equations?

Try `v1#branch` it represents the source current. I don't know if this works with 50-Ohm ports. 

> I probably could have used the old S-parameter probes from years ago but didn't...

S-probes should be never used in new designs.  

I am transferring this issue to discussions https://github.com/ra3xdh/qucs_s/discussions

**Comment by tomhajjar** [comment]
Holger said to use .PARAM Z0=50, .CSPARAM Z0='Z0'. 

Is there a way to do this in Qucs-S since no .CSPARAM exists? 

I tried using .INCLUDE SCRIPT and .GLOBAL PARAM but in both cases Qucs-S crashes... 

https://sourceforge.net/p/ngspice/discussion/120973/thread/2d9f28fa73/

![2024-02-16_132950](https://github.com/ra3xdh/qucs_s/assets/102838604/a2b32fda-1194-43cb-a0da-cf3b4650c1bb)

[S-parameters_ngspice_AC_prj.zip](https://github.com/ra3xdh/qucs_s/files/14314715/S-parameters_ngspice_AC_prj.zip)

**Comment by ra3xdh** [comment]
> Is there a way to do this in Qucs-S since no .CSPARAM exists?

Type everything in the INCLUDE SCRIPT window. You may use whatever directives you want with it.

I don't understand what are you trying to achieve and why. The S-parameters should not be extracted to from AC simulation. This method presents only academic interest. Please find another way to convert this project.
